### PR TITLE
Add possibility to refer environment variables in properties

### PR DESCRIPTION
**Pull request by TatsianaLuchonak**

### CODE REVIEW

File: vividus/src/main/java/org/vividus/configuration/ConfigurationResolver.java [hunk 1]
```diff
@@ -114,7 +115,14 @@ public static ConfigurationResolver getInstance() throws IOException
             String key = (String) entry.getKey();
             String value = (String) entry.getValue();
             deprecatedPropertiesHandler.warnIfDeprecated(key, value);
-            entry.setValue(propertyPlaceholderHelper.replacePlaceholders(value, properties::getProperty));
+            if (key.startsWith(ENVIRONMENT_VARIABLES))
```
Comment (valfirst) [inline vividus/src/main/java/org/vividus/configuration/ConfigurationResolver.java]: I think any property can refer environment variable, not only global variables

File: vividus/src/main/java/org/vividus/configuration/ConfigurationResolver.java [hunk 2]
```diff
@@ -210,6 +218,16 @@ private static void processSystemProperties(Properties properties)
         }
     }
 
+    private static void processEnvironmentVariable(Entry<Object, Object> entry, String value)
+    {
+        String variableName = StringUtils.substringBetween(value, PLACEHOLDER_PREFIX, PLACEHOLDER_SUFFIX);
```
Comment (valfirst) [inline vividus/src/main/java/org/vividus/configuration/ConfigurationResolver.java]: multiple environment variables can be referred in properties:
```properties
web-application.main-page-url=${APP_SHEME}://${APP_HOST}:${APP_PORT}/
```

File: vividus-tests/src/main/resources/properties/environment/integration/envinroment.properties [hunk 3]
```diff
@@ -1,3 +1,6 @@
 web-application.main-page-url=http://example.org/
 
 bdd.variables.global.encrypted-variable=ENC(tUWY/ltftWUA5tDnlHU+jrmBi1iuWqge)
+bdd.variables.global.java=${JAVA_HOME}
```
Comment (ikalinin1) [inline vividus-tests/src/main/resources/properties/environment/integration/envinroment.properties]: Whats are guaranteeing there is JAVA_HOME  variable?

File: vividus/src/main/java/org/vividus/configuration/ConfigurationResolver.java [hunk 4]
```diff
@@ -99,6 +99,7 @@ public static ConfigurationResolver getInstance() throws IOException
 
         Properties overridingAndSystemProperties = new Properties();
         overridingAndSystemProperties.putAll(overridingProperties);
+        overridingAndSystemProperties.putAll(System.getenv());
```
Comment (ikalinin1) [inline vividus/src/main/java/org/vividus/configuration/ConfigurationResolver.java]: @valfirst  Why we don't have unit tests for the class?

Comment (valfirst) [reply vividus/src/main/java/org/vividus/configuration/ConfigurationResolver.java]: it contains a lot of logic which includes interaction with system resources and etc. the logic is covered by integration tests: https://github.com/vividus-framework/vividus/blob/master/vividus/src/integrationTest/java/org/vividus/configuration/BeanFactoryTests.java

File: vividus-tests/src/main/resources/properties/environment/integration/envinroment.properties (hunk 3 above)
Comment (valfirst) [reply vividus-tests/src/main/resources/properties/environment/integration/envinroment.properties]: no guarantee at all, just high probability

**Pull request merged**